### PR TITLE
Remove enrolment feature

### PR DESCRIPTION
**Pull request by tijmenb**
## Context

We're removing the "enrolled" state because it won't be used in this cycle, and gives us a clean slate to build the feature next cycle. 

## Changes proposed in this pull request

Remove the state and associated code. We don't generate test applications in this state anymore since https://github.com/DFE-Digital/apply-for-teacher-training/pull/2700, and I've manually removed existing enrolled applications from QA, Sandbox and Staging. This should be safe to merge. 

## Guidance to review

Did I miss anything?

## Link to Trello card

https://trello.com/c/1DdTHKQI/2604-remove-enrolment-state

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

**Comment by duncanjbrown** [comment]
Looks pretty good to me! 

- Not sure what to do with the endpoint — `410` is cool but it's a hard removal rather then deprecation, which is technically a breaking change
- Enrolment still cropping up in:
  - `provider_interface/filter_component_spec.rb`
  - `status_box_component_preview.rb`
  - `filter_component_preview.rb`

**Comment by duncanjbrown** [comment]
I'm adopting this PR

File: app/services/support_interface/application_monitor.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@ def applications_to_courses_with_sites_without_vacancies
   private
 
     def active_applications
-      ApplicationChoice.where('status NOT IN (?)', %w[rejected declined withdrawn enrolled conditions_not_met])
+      ApplicationChoice.where('status NOT IN (?)', ApplicationStateChange::TERMINAL_STATES)
```
Comment (mnacos) [inline app/services/support_interface/application_monitor.rb:30]: :+1:

**Review by mnacos — COMMENTED** [review]
Looking good! I think changing the 410 into a noop that notifies us makes sense for operational reasons.

Quick question about HESA ITT data... Should the data be shown when the application is in 'recruited' state now?

See https://github.com/DFE-Digital/apply-for-teacher-training/blob/remove-enrolment/app/presenters/vendor_api/single_application_presenter.rb#L48

Otherwise, this PR is ready to go from a code perspective. Some copy will need to be updated either as part of this PR or a separate story. In particular:

Our API docs still mention enrollment:

- `app/views/api_docs/pages/home.md`
- `app/views/api_docs/pages/usage.md`

Some public-facing documents still mention enrollment:

- `app/views/candidate_interface/equality_and_diversity/start.html.erb`
- `app/views/content/privacy_policy.md`
- `app/views/provider_interface/provider_agreements/data_sharing_agreement.html.erb`
- `app/views/api_docs/pages/alpha_release_notes.md`

The same applies to these emails:

- `app/views/candidate_mailer/application_submitted.text.erb`
- `app/views/candidate_mailer/conditions_met.text.erb`

---

This PR also removes a few things related to the Covid-19 banner, which I haven't tested.

**Comment by duncanjbrown** [comment]
thanks for the review @mnacos — will fix up tomorrow!

**Comment by duncanjbrown** [comment]
> Quick question about HESA ITT data... Should the data be shown when the application is in 'recruited' state now?

Yes, probably — I think we can make the decision when we put it in.

> Our API docs still mention enrollment:
> 
> * `app/views/api_docs/pages/home.md`
> * `app/views/api_docs/pages/usage.md`

Thank you — I've removed mention of enrolment from both these docs.

> 
> Some public-facing documents still mention enrollment:
> 
> * `app/views/candidate_interface/equality_and_diversity/start.html.erb`

I replaced "enrolled on a course" with "successfully completed your application".

> * `app/views/content/privacy_policy.md`

I left this alone as it doesn't necessarily imply that we have an `enrolment` feature: 

"Once a candidate has enrolled with a teacher training provider, we will not be able to ask them to remove personal data from their systems. Please contact the provider separately."

> * `app/views/provider_interface/provider_agreements/data_sharing_agreement.html.erb`

I left this alone as I don't want to touch the DSA — it's currently under review and I've added a note to https://trello.com/c/mkwDcb2R/2684-draft-dsa-changes

> * `app/views/api_docs/pages/alpha_release_notes.md`

This is a historical document and doesn't need to change.

> 
> The same applies to these emails:
> 
> * `app/views/candidate_mailer/application_submitted.text.erb`

Removed the "up until enrolment" clause.

> * `app/views/candidate_mailer/conditions_met.text.erb`

"If they have not already, $PROVIDER should be in contact about enrolling on the course. You can contact them directly if you have any questions about this process" doesn't imply we have an enrolment feature, so left alone.

File: app/services/application_state_change.rb [hunk 2]
```diff
@@ -2,11 +2,13 @@ class ApplicationStateChange
   include Workflow
 
   STATES_NOT_VISIBLE_TO_PROVIDER = %i[unsubmitted awaiting_references application_complete cancelled].freeze
```
[the remark this answers is not in the document]
Comment (duncanjbrown) [reply app/services/application_state_change.rb:7]: It is useful in the TAD spec — maybe we could refactor next time we visit that?

**Review by mnacos — APPROVED** [review]
:+1:

**Pull request merged**